lib/licensed/sources/swift.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+require "json"
+require "pathname"
+require "uri"
+
+require "licensed/sources/helpers/content_versioning"
+
+module Licensed
+  module Sources
+    class Swift < Source
+      def enabled?
+        return unless Licensed::Shell.tool_available?("swift") && swift_package?
+        File.exist?(package_resolved_file_path)
+      end
+
+      def enumerate_dependencies
+        pins.map { |pin|
+          Dependency.new(
+            name: pin["package"],
+            path: dependency_path_for_url(pin["repositoryURL"]),
```

Comment:
@jonabc Following up from our [previous discussion about error handling](https://github.com/github/licensed/pull/363#discussion_r636550443), what's the correct way to handle a possible `URI::InvalidURIError` or `ArgumentError` from `dependency_path_for_url`? Should I return a `Dependency` with a nil path and an `errors` value, or return `nil` to skip it?

Reply:
> Should I return a Dependency with a nil path and an errors value

👍  I think this is the best option, but it would be good to verify these error cases with tests if possible